### ISSUE TITLE
Timeout in OSS-Fuzz

### ISSUE THREAD

**Issue by kevinbackhouse** [Exiv2/exiv2#2952]
OSS-Fuzz has found a "timeout" issue: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=66957
It's not a security bug, because it's not reproducible in Exiv2. It only affects the fuzz target that we run in OSS-Fuzz.

To reproduce the issue, build the fuzz target and then run it like this:

```bash
./bin/fuzz-read-print-write poc.jpg
```

poc file:  ![Image](https://github.com/Exiv2/exiv2/assets/4358136/ca1c18f9-3bb0-4f3d-b84a-9e81b768093a)

It runs for a long time, because these calls to `md.print()` are generating very large strings:

https://github.com/Exiv2/exiv2/blob/77915ad17b103793017e2b3b5ffd6c8a3ab1e916/fuzz/fuzz-read-print-write.cpp#L22-L26

The reason why Exiv2 _isn't_ affected is because it has this code:

https://github.com/Exiv2/exiv2/blob/77915ad17b103793017e2b3b5ffd6c8a3ab1e916/app/actions.cpp#L453-L455

In other words, Exiv2 doesn't print anything if the tag name starts with "0x". I could easily fix the fuzz target by adding similar code, but I'd quite like to understand what that code is doing. Does anybody know the reason for it?
